modules/reindex/src/main/java/org/elasticsearch/index/reindex/Reindexer.java

```diff
@@ -265,7 +266,11 @@ protected void finishHim(Exception failure, List<BulkItemResponse.Failure> index
              */
             index.routing(mainRequest.getDestination().routing());
             index.setPipeline(mainRequest.getDestination().getPipeline());
-            // OpType is synthesized from version so it is handled when we copy version above.
+            // Only copy the op_type if it has been set to CREATE.
+            // (INDEX is the default and UPDATE and DELETE are invalid values)
+            if (mainRequest.getDestination().opType() == DocWriteRequest.OpType.CREATE) {
```

Comment:
According to the reindex docs, op_type is a valid argument. However it was currently not supported, as the op_type wasn't copied over from the destination index request template to the actual index requests that reindex api generates.

Reply:
That is true. I will remove it and include the it in the commit message.